cmd/airflow.go

```diff
@@ -162,6 +168,50 @@ func newAirflowInitCmd() *cobra.Command {
 	return cmd
 }
 
+func newAirflowUpgradeTestCmd(astroClient astro.Client) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "upgrade-test",
+		Short:   "Run tests to see if your environment and DAGs are compatable with a new version of Airflow or Runtime",
+		Long:    "Run some tests to see if your environment and DAGs are compatable with a new version of Airflow or Runtime",
+		Example: initCloudExample,
+		// ignore PersistentPreRunE of root command
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return airflowUpgradeTest(cmd, astroClient)
+		},
+	}
+	cmd.Flags().StringVarP(&airflowVersion, "airflow-version", "a", "", "Version of Airflow you want to upgrade to. If not specified, latest is assumed.")
+	cmd.Flags().BoolVarP(&conflictTest, "conflict-test", "d", false, "dependency conflict test only")
+	cmd.Flags().BoolVarP(&versionTest, "version-test", "", false, "dependency version comparison only")
+	cmd.Flags().BoolVarP(&dagTest, "dag-test", "", false, "dag test only")
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "i", "", "ID of Deployment your want check dependency versions against. Using ")
+	var err error
+	var avoidACFlag bool
+
+	// In case user is connected to Astronomer Platform and is connected to older version of platform
+	if context.IsCloudContext() || houstonVersion == "" || (!context.IsCloudContext() && houston.VerifyVersionMatch(houstonVersion, houston.VersionRestrictions{GTE: "0.29.0"})) {
+		cmd.Flags().StringVarP(&runtimeVersion, "runtime-version", "v", "", "Specify a version of Astro Runtime that you want to create an Astro project with. If not specified, the latest is assumed. You can change this version in your Dockerfile at any time.")
+	} else { // default to using AC flag, since runtime is not available for these cases
+		useAstronomerCertified = true
+		avoidACFlag = true
+	}
+
+	if !context.IsCloudContext() && !avoidACFlag {
+		cmd.Example = initSoftwareExample
+		cmd.Flags().BoolVarP(&useAstronomerCertified, "use-astronomer-certified", "", false, "If specified, specify an Astronomer Certified Airflow image to upgrade to instead of Astro Runtime.")
+	}
+
+	_, err = context.GetCurrentContext()
+	if err != nil && !avoidACFlag { // Case when user is not logged in to any platform
+		cmd.Flags().BoolVarP(&useAstronomerCertified, "use-astronomer-certified", "", false, "If specified, specify an Astronomer Certified Airflow image to upgrade to instead of Astro Runtime.")
```

Comment:
```suggestion
		cmd.Flags().BoolVarP(&useAstronomerCertified, "use-astronomer-certified", "", false, "Use an Astronomer Certified image instead of Astro Runtime. Use the airflow-version flag to specify your AC version.)
```